homeassistant/components/aqara/__init__.py

```diff
@@ -0,0 +1,205 @@
+"""Support for Aqara Smart devices."""
+from __future__ import annotations
+
+import logging
+from typing import NamedTuple
+
+from aqara_iot import (  # AqaraDevice,
+    AqaraDeviceListener,
+    AqaraDeviceManager,
+    AqaraHomeManager,
+    AqaraOpenAPI,
+    AqaraOpenMQ,
+    AqaraPoint,
+)
+import requests
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.dispatcher import async_dispatcher_connect, dispatcher_send
+
+from .const import (
+    AQARA_DISCOVERY_NEW,
+    AQARA_HA_SIGNAL_REGISTER_POINT,
+    AQARA_HA_SIGNAL_UPDATE_ENTITY,
+    AQARA_HA_SIGNAL_UPDATE_POINT_VALUE,
+    CONF_AUTH_TYPE,
+    CONF_COUNTRY_CODE,
+    CONF_PASSWORD,
+    CONF_PROJECT_TYPE,
+    CONF_USERNAME,
+    DOMAIN,
+    PLATFORMS,
+)
+from .util import string_dot_to_underline, string_underline_to_dot
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class HomeAssistantAqaraData(NamedTuple):
+    """Aqara data stored in the Home Assistant data object."""
+
+    device_listener: DeviceListener  # AqaraDeviceListener
+    device_manager: AqaraDeviceManager
+    home_manager: AqaraHomeManager
+    aqara_mqtt_client: AqaraOpenMQ
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Async setup hass config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    # Project type has been renamed to auth type in the upstream Aqara IoT SDK.
+    # This migrates existing config entries to reflect that name change.
+    if CONF_PROJECT_TYPE in entry.data:
+        data = {**entry.data, CONF_AUTH_TYPE: entry.data[CONF_PROJECT_TYPE]}
+        data.pop(CONF_PROJECT_TYPE)
+        hass.config_entries.async_update_entry(entry, data=data)
```

Comment:
This looks copied from the Tuya integration and seems unlikely that this applies to Aqara?